dotCMS/src/main/java/com/dotmarketing/util/ImportFileParams.java

```diff
@@ -0,0 +1,169 @@
+package com.dotmarketing.util;
+
+import com.dotcms.repackage.com.csvreader.CsvReader;
+import com.liferay.portal.model.User;
+import java.util.function.LongConsumer;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import org.immutables.value.Value;
+
+/**
+ * Abstract base class for file import parameters.
+ * This class will be processed by the @Value.Immutable annotation to generate
+ * an immutable implementation with a Builder pattern.
+ */
+@Value.Immutable
+@Value.Style(
+        // Style configuration for code generation
+        builderVisibility = Value.Style.BuilderVisibility.PUBLIC, // Public builder
+        allParameters = true, // Constructor with all parameters
+        defaults = @Value.Immutable(copy = false) // Don't generate copy methods
+)
+public abstract class ImportFileParams {
+
+    /**
+     * @return Import ID
+     */
+    @Value.Parameter
+    public abstract Long importId();
+
+    /**
+     * @return Current site ID
+     */
+    @Value.Parameter
+    public abstract String siteId();
+
+    /**
+     * @return Content type inode
+     */
+    @Value.Parameter
+    public abstract String contentTypeInode();
+
+    /**
+     * @return array of key fields
+     */
+    @Value.Parameter
+    public abstract String[] keyFields();
```

Comment:
why not a set would be less open to errors

Reply:
compatibility wiht the old methods